android-components/components/browser/engine-gecko/src/main/java/mozilla/components/experiment/NimbusExperimentDelegate.kt

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.experiment
+
+import mozilla.components.browser.engine.gecko.GeckoNimbus
+import mozilla.components.support.base.log.logger.Logger
+import org.json.JSONObject
+import org.mozilla.geckoview.ExperimentDelegate
+import org.mozilla.geckoview.GeckoResult
+
+/**
+ * Default Nimbus [ExperimentDelegate] implementation to communicate with mobile Gecko and GeckoView.
+ */
+class NimbusExperimentDelegate : ExperimentDelegate {
+
+    private val logger = Logger(NimbusExperimentDelegate::javaClass.name)
+
+    /**
+     * Retrieves experiment information on the feature for use in GeckoView.
+     *
+     * @param feature Nimbus feature to retrieve information about
+     * @return a JSON object containing experiment information
+     */
+    override fun onGetExperimentFeature(feature: String): GeckoResult<JSONObject>? {
+        val result = GeckoResult<JSONObject>()
+        val nimbusFeature = GeckoNimbus.getFeature(feature)
+        if (nimbusFeature != null) {
+            result.complete(nimbusFeature.toJSONObject())
+        } else {
+            logger.warn("Could not find Nimbus feature '$feature' to retrieve experiment information.")
+            result.complete(null)
+        }
+        return result
+    }
+
+    /**
+     * Records that an exposure event occurred with the feature.
+     *
+     * @param feature Nimbus feature to record information about
+     * @return a [GeckoResult] with true if the feature was found and recorded, false if not found
+     */
+    override fun onRecordExposureEvent(feature: String): GeckoResult<Boolean>? {
+        val result = GeckoResult<Boolean>()
+        val nimbusFeature = GeckoNimbus.getFeature(feature)
```

Comment:
If a user has opted out of experiments, does `GeckoNimbus` already know of the opt-in/opt-out and handles it internally or do I need to check additionally here as well?

Reply:
Answer: No, you don't have to do anything extra here: the Nimbus SDK has taken care of that.